src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1140,4 +1143,18 @@ def download_url_for_file_in_repo(project, package_name, repository, architectur
     # only use API for logged in users if the mirror is not available - return nil otherwise
     rpm_url(project, package_name, repository.name, architecture, filename) unless User.current.is_nobody?
   end
+
+  def require_repository
+    @repository = @project.repositories.find_by(name: params[:repository])
+    return if @repository
```

Comment:
you could return @repository.name, so that you don't need to change the code everywhere to add the `.name` :thinking: